apps/root/src/pages/earn-access-now/components/elegibility-confirmation/use-cases.tsx

```diff
@@ -0,0 +1,242 @@
+import Address from '@common/components/address';
+import { ElegibilityConfirmationStatus } from '.';
+
+const TitleContainer = styled(ContainerBox).attrs({
+  flexDirection: 'column',
+  gap: 2,
+})``;
+
+const StyledTitle = styled(Typography).attrs({ variant: 'h3Bold', textAlign: 'center' })`
+  color: ${({ theme }) => colors[theme.palette.mode].typography.typo1};
+`;
+
+const StyledDescription = styled(Typography).attrs({ variant: 'bodyRegular', textAlign: 'center', maxWidth: '40ch' })`
+  color: ${({ theme }) => colors[theme.palette.mode].typography.typo2};
+`;
+
+const StyledViewableButton = styled(Button).attrs({ variant: 'contained' })`
+  background-color: ${({ theme }) => colors[theme.palette.mode].accent.accent600};
+  cursor: default;
+  pointer-events: none;
+`;
+
+const ClaimEarnButton = () => {
+  const pushToHistory = usePushToHistory();
+
+  const onEnterEarnNow = () => {
+    pushToHistory(`/${EARN_ROUTE.key}`);
+  };
+
+  return (
+    <Button variant="contained" onClick={onEnterEarnNow}>
+      <FormattedMessage
+        description="earn-access-now.eligibility.confirmation.eligible.button"
+        defaultMessage="Enter Earn Now!"
+      />
+    </Button>
+  );
+};
+
+export const LoadingCase = () => (
+  <>
+    <CenteredLoadingIndicator size={55} />
+    <StyledViewableButton>
+      <FormattedMessage
+        description="earn-access-now.eligibility.confirmation.loading.button"
+        defaultMessage="Checking your eligibility..."
+      />
+    </StyledViewableButton>
+  </>
+);
+
+export const NotEligibleStatus = () => {
+  const openConnectWalletModal = useOpenConnectModal();
+  const trackEvent = useTrackEvent();
+
+  const onConnectWallet = React.useCallback(() => {
+    openConnectWalletModal(WalletActionType.link);
+    trackEvent('Earn Early Access - Elegibility confirmation - Link wallet');
+  }, [openConnectWalletModal, trackEvent]);
+  return (
+    <>
+      <TitleContainer>
+        <StyledTitle>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.not-eligible.title"
+            defaultMessage="You're Not Eligible 😔"
+          />
+        </StyledTitle>
+        <StyledDescription>
+          <FormattedMessage
+            description="earn-access-now.eligibility.confirmation.not-eligible.description"
+            defaultMessage="It seems none of your wallets meet the criteria for Earn Early Access. Don't worry! You can still join by trying another wallet or finding a referral code on our Discord or Twitter!"
+          />
+        </StyledDescription>
+      </TitleContainer>
+      <Button variant="outlined" onClick={onConnectWallet}>
+        <FormattedMessage
+          description="earn-access-now.eligibility.confirmation.not-eligible.button"
+          defaultMessage="Try with another wallet"
+        />
+      </Button>
+    </>
+  );
+};
+
+export const ElegibleCase = () => {
+  // TODO: Specify elegible wallet address if isAuth
```

Comment:
what do we need this for?

Reply:
I think it would be useful for the user to know which wallet address is elegible, instead of saying 'Your wallet meets the criteria'